stackstac/rio_reader.py

```diff
@@ -304,6 +304,7 @@ def __init__(
         dtype: np.dtype,
         fill_value: Union[int, float],
         rescale: bool,
```

Comment:
It feels a little weird to have both `rescale` and `scale_offset`. We should probably just remove `rescale`, and at the same time, remove support for rescaling from metadata in the COG. As mentioned in https://github.com/gjoseph92/stackstac/issues/63#issuecomment-868726296, this doesn't seem to get used often anyway, and makes things more complex (it's unclear which should take precedence).

Then, for `stack(..., rescale=False)`, we'd just pass a no-op `scale_offset` of `(1, 0)`.

Reply:
I have made the recommended changes, happy to get feedback to see if it makes sense how I use `DEFAULT_SCALE, DEFAULT_OFFSET` across the repo.